docs/decisions/0003-django-setting-toggles.rst

```diff
@@ -0,0 +1,66 @@
+Django Setting Toggles
+======================
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Django Settings are implicitly defined when they are used.  For example, in ``edx-platform`` the ``FEATURES`` setting dict is used, and an example usage might look like::
+
+    settings.FEATURES.get('EXAMPLE_FEATURE', False)
+
+There are several problems with this style of implicit definition:
+
+* Each usage may or may not define a default value, and the default values are not guaranteed to be consistent.
+* It is unclear when and if a default value should be provided with the usage.
+* If there are multiple uses of a setting, it unclear where the code annotation that documents the toggle should live.
+* Django Settings toggles are unable to report the same level of detail in the toggle state report as Waffle toggles, because they can't be queried for location or code ownership information.
+
+Decision
+--------
+
+We will introduce a simple wrapping class that can be used to explicitly define a Django Setting toggle.
+
+Here is an example declaration::
+
+    EXAMPLE_FEATURE_TOGGLE = DjangoSettingToggle(
+        "FEATURES['EXAMPLE_FEATURE']",
+        default_value=False,
+    )
+
+It's usage can now match that of the waffle wrapping classes (e.g. WaffleFlag).
+
+Here is an example usage::
+
+    EXAMPLE_FEATURE_TOGGLE.is_enabled()
```

Comment:
Note that this API works only for boolean settings (i.e: feature flags).

Reply:
Yes. It is specifically designed for toggles.